src/software/geom/angle_map.h

```diff
@@ -0,0 +1,58 @@
+#include <vector>
+
+#include "software/geom/angle_segment.h"
+
+/**
+ * Represents an AngleMap that is confined to a top and bottom angle, with angles
+ * described as going from pi -> 0 -> -pi
+ */
+class AngleMap
+{
+   public:
+    /**
+     * Constructs an AngleMap with a specified top angle, bottom angle, and max number of
+     * possible occupied AngleSegments within this map
+     *
+     * @param top_angle the top angle (most north) of the AngleSegment this map occupies
+     * @param bottom_angle the bottom angle (most south) of the AngleSegment this map
+     * occupies
+     * @param max_num_obstacles the max number of possible occupied AngleSegments within
+     * this map
+     */
+    AngleMap(Angle top_angle, Angle bottom_angle, size_t max_num_obstacles);
```

Comment:
I think max_num_obstacles should be more like reserved num obstacles (since AngleMap supports adding more than max_num_obstacles I asssume), and set a default arg to 1. I get that doing the initial reservation is good for performance but it's annoying to set for users that don't need the performance

Reply:
done